services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -279,17 +283,24 @@ public void onPartitionStatusChange(String topic, ReadOnlyPartitionStatus partit
    * @param nodeId
    * @return
    */
-  protected static double getNodeResponsibilityForQuota(PartitionAssignment partitionAssignment, String nodeId) {
+  protected static double getNodeResponsibilityForQuota(
+      HelixCustomizedViewOfflinePushRepository customizedViewRepository,
+      PartitionAssignment partitionAssignment,
+      String nodeId) {
     double thisNodePartitionPortion = 0; // 1 means full responsibility for serving a partition. 0.33 means serving 1 of
-                                         // 3 replicas for a partition
+    // 3 replicas for a partition
     for (Partition p: partitionAssignment.getAllPartitions()) {
-      // TODO: getReadyToServeInstances is deprecated and no longer works with L/F. We need to either
```

Comment:
For my understanding: I thought that we are moving from EV to CV because CV has more apt counts of replicas (i,e: COMPLETED). As per these comments, it looks like the reason is much more involved. can you please explain why `getReadyToServeInstances ` no longer works with L/F? Basically why we are moving away from EV even as both `onExternalViewChange()` and `onCustomizedViewChange()` seems to call `getNodeResponsibilityForQuota`

Reply:
In the only O/B/O model EV means everything. i.e. If it's ONLINE in EV it means it's ready to serve and if it's OFFLINE or BOOTSTRAPPING it means it's not ready. In L/F model the information on EV is incomplete. Leader or STANDBY doesn't necessarily mean if it's ready to serve or not. We need to check CV to make sure it's COMPLETED. `getNodeResponsibilityForQuota` is called on both EV and CV changes because EV changes like going from STANDBY/LEADER->OFFLINE will require quota to be redistributed since we are missing a replica. Vice versa if a new replica is bootstrapped/restarted and ready to serve again the quota also need to be adjusted.